modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -1120,6 +1155,94 @@ private float getTabAdvance() {
+     *   following a white space sequence)
+     *
+     * White space collapsing:
+     *
+     * Only white space that is present at soft wrapped locations is collapsed to
+     * zero. Any other white space is preserved. This includes white space between
+     * words, leading and trailing white space, and white space around hard wrapped
+     * locations.
+     *
+     * Alignment:
+     *
+     * The alignment calculation only looks at the width of all the significant
+     * characters in each line. Significant characters are any non white space
+     * characters and any white space that has been preserved (white space that wasn't
+     * collapsed due to soft wrapping).
+     *
+     * Alignment does not take text effects, such as strike through and underline, into
+     * account. This means that such effects can appear unaligned. Trailing spaces at a
+     * soft wrap location (that are underlined for example), may show the underline go
+     * outside the logical bounds of the text.
+     *
+     * Example, where <SW> indicates a soft wrap location, and <LF> is a line feed:
+     *
+     *     "   The   quick <SW>brown fox jumps <SW> over the <LF> lazy dog   "
+     *
+     * Would be rendered as (left aligned):
+     *
+     *     "   The   quick"
+     *     "brown fox jumps"
+     *     "over the "
+     *     " lazy dog   "
+     *
+     * The alignment calculation uses the above bounds indicated by the double
+     * quotes, and so right aligned text would look like:
+     *
+     *      "   The   quick"
+     *     "brown fox jumps"
+     *           "over the "
+     *        " lazy dog   "
+     *
+     * Note that only the white space at the soft wrap locations is collapsed.
+     * In all other locations the space was preserved (the space between words
+     * where no soft wrap occurred, the leading and trailing space, and the
+     * space around the hard wrapped location).
+     *
+     * Text effects have no effect on the alignment, and so with underlining on
+     * the right right aligned text would look like:
+     *
+     *      "___The___quick_"     (one collapsed space becomes visible here)
+     *     "brown_fox_jumps__"    (two collapsed spaces become visible here)
+     *           "over_the_"
+     *        "_lazy_dog___"
+     *
+     * Note that text alignment has not changed at all, but the bounds are exceeded
+     * in some locations to allow for the underline. Controls displaying such texts
+     * will likely clip the underlined parts exceeding the bounds.
+     *
+     * Users wishing to mitigate some of these perhaps surprising results can ensure
+     * they use trimmed texts, and avoid the use of line breaks, or at least ensure
+     * that line breaks are not preceded or succeeded by white space (activating
+     * line wrapping is not equivalent to collapsing any consecutive white space
+     * no matter where it occurs).
+     */
+
```

Comment:
@karthikpandelu could you please take a look at this comment here and this PR in general (since you both are modifying the same file)?

Is there anything to add?